[dagster-airflow] persistent db docs (#12485)
dagster-io/dagster: docs/content/integrations/airflow/migrating-to-dagster.mdx

---

> ## Deciding on persistent vs ephemeral airflow database

Reviewer: Should this be considered a prerequisite (something the user needs to do before they start) or a step in the tutorial itself? Right now this section is sorta hanging out between the Prereqs and the first step, which is a bit odd.

It looks like it should be a step, based on the fact that the user is writing some code and interacting with the `dagster-airflow` library. If that's the case, we'll need to shuffle the order of steps around a bit.

Author: 👍 great callout.

I moved this into a step right before going to production.

> The dagster airflow migration tooling supports two methods for persisting airflow metadatabase state. By default it will us an ephemeral database that is scoped to every job run and thrown away as soon a job run terminates. You can also configure a persistent database that will be shared by all job runs. This tutorial uses the ephemeral database but you will want to use the persistent database option if you need the following features:

Reviewer: ```suggestion
The Dagster Airflow migration tooling supports two methods for persisting Airflow metadatabase state. By default, it will use an ephemeral database that is scoped to every job run and thrown away as soon as a job run terminates. You can also configure a persistent database that will be shared by all job runs. This tutorial uses the ephemeral database, but the persistent database option is recommended if you need the following features:
```

- `retry-from-failure` support in Dagster

> - passing airflow state between dag runs (ie xcoms)

Reviewer: ```suggestion
- Passing Airflow state between DAG runs (i.e., xcoms)
```

- `SubDAGOperators`
- Airflow Pools

> If you have complex airflow DAGs you will likely want to have a persistent database when making your migration.

Reviewer 2: More concrete examples would be good here. (e.g. uses xcoms and executes across nodes)

> If you have complex airflow DAGs (cross-dag state sharing, pools) or the ability to retry-from-failure you will need to have a persistent database when making your migration.

Reviewer: Is `retry-from-failure` the name of a feature or a literal value you'd see in code?

Author: neither, it's an attempt to describe when a multi-op job has failed and you want to retry from the op that failed, not retry/rerun the entire job and ops that had succeeded

> You can configure your persistent airflow database by providing an `airflow_db` to the resource_defs param of the dagster-airflow api's.

Reviewer: ```suggestion
You can configure your persistent airflow database by providing an `airflow_db` to the `resource_defs` parameter of the `dagster-airflow` APIs:
```

---

> ## Step 6: Deciding on persistent vs ephemeral airflow database

Reviewer: ```suggestion
## Step 6: Deciding on persistent vs ephemeral Airflow database
```

The Dagster Airflow migration tooling supports two methods for persisting Airflow metadatabase state. By default, it will use an ephemeral database that is scoped to every job run and thrown away as soon as a job run terminates. You can also configure a persistent database that will be shared by all job runs. This tutorial uses the ephemeral database, but the persistent database option is recommended if you need the following features:

- `retry-from-failure` support in Dagster
- Passing Airflow state between DAG runs (i.e., xcoms)
- `SubDAGOperators`
- Airflow Pools

> If you have complex airflow DAGs (cross-dag state sharing, pools) or the ability to retry-from-failure you will need to have a persistent database when making your migration.

Reviewer: ```suggestion
If you have complex Airflow DAGs (cross-DAG state sharing, pools) or the ability to retry-from-failure you will need to have a persistent database when making your migration.
```

❓ Should `retry-from-failure` be formatted as code? It is in the list preceding this paragraph - either way, we should be consistent

> You can configure your persistent airflow database by providing an `airflow_db` to the `resource_defs` parameter of the `dagster-airflow` APIs:

Reviewer: ```suggestion
You can configure your persistent Airflow database by providing an `airflow_db` to the `resource_defs` parameter of the `dagster-airflow` APIs:
```

> ---

Reviewer: I can't comment on unchanged lines, but in line 216, could you change `dags` to `DAGs`? As this is an acronym, it should be in all caps.
